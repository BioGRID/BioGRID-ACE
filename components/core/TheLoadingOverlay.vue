<template>
    <div id="ace-loading-overlay">
        <v-overlay :value="loadingOverlayVisible">
            <v-progress-circular
                indeterminate
                color="yellow accent-4"
                size="120"
                width="10"
            >
                Loading
            </v-progress-circular>
        </v-overlay>
    </div>
</template>

<script lang="ts">
import { Component, Vue, State } from 'nuxt-property-decorator'

@Component
export default class TheLoadingOverlay extends Vue {
    @State private loadingOverlayVisible!: boolean;
}
</script>
