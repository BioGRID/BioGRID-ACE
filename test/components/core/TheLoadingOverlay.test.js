// LIBRARIES
import Vuetify from 'vuetify'
import Vuex from 'vuex'

// COMPONENTS
import TheLoadingOverlay from '@/components/core/TheLoadingOverlay.vue'

// UTILITIES
import { shallowMount, createLocalVue } from '@vue/test-utils'

// SETUP LOCALIZED VUE
const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper = null

// RUN BEFORE EACH TEST
beforeEach(() => {
    const vuetify = new Vuetify()
    const store = new Vuex.Store({
        state: {
            loadingOverlayVisible: false
        }
    })
    wrapper = shallowMount(TheLoadingOverlay, {
        localVue,
        vuetify,
        store
    })
})

// RUN AFTER EACH TEST
afterEach(() => {
    wrapper.destroy()
})

describe('components/core/TheLoadingOverlay', () => {
    test('renders', () => {
        expect(wrapper.exists()).toBe(true)
    })
    test('renders properly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('defaults to invisible', () => {
        expect(wrapper.vm.loadingOverlayVisible).toBe(false)
    })
})
