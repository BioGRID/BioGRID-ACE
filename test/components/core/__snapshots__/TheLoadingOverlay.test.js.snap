// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/core/TheLoadingOverlay renders properly 1`] = `
"<v-overlay-stub color=\\"#212121\\" dark=\\"true\\" opacity=\\"0.46\\" zindex=\\"5\\" id=\\"ace-loading-overlay\\">
  <v-progress-circular-stub color=\\"yellow accent-4\\" indeterminate=\\"true\\" rotate=\\"0\\" size=\\"120\\" width=\\"10\\" value=\\"0\\">
    Loading
  </v-progress-circular-stub>
</v-overlay-stub>"
`;
